### PR TITLE
Players endpoint and cleanup

### PR DESCRIPTION
**Pull request by kevinhughes27**
This PR does a lot (please forgive the large diff) but it is also a huge improvement to the structure of the code and the APIs. 

### Main Changes
On the backend the main addition is the `/api/players` endpoint which returns an array of players. The player's name, team name and salary are serialized. The team dashboard and trade simulator views only care about this information (no stats) and they are only ever useful in the current context (aka the week picker is not useful and in face is always the same now because of how salary is calculated and stored). By adding and using this new endpoint I was able to remove significant complexity from the `build_stats_response` function which was getting out of hand. The android app could also use this API letting us further simplify the `/api/teams` endpoint.

I made the changes to the frontend to consume the new players API which let me remove the Stats.js store (which sort of provided the players API to the frontend before). I also refactored out a bunch of smaller components.

### Other Changes
* added caching for all api endpoints forever. We can always restart dynos if we need to clear the cache manually. It clears itself on both operations that effect it right now.
* removed the rest of the `flow` syntax since we stopped type checking a while ago
* adds a few more stats to leaderboards
* change the `All` tab to show cumulative `pay` and not salary

@keatesc will this affect the spreadsheet sync script?

### CODE REVIEW

**Comment by kevinhughes27** [comment]
This will break the TeamDashboard and TradeSimulator until the client does a hard refresh because of the service worker.

File: client/src/Components/TradeSimulator/index.js [hunk 1]
```diff
@@ -1,54 +1,30 @@
...
+      trades: [],
+      playerA: '',
+      playerB: '',
     }
   }
 
+  componentWillMount() {
+    fetch('/api/players')
+      .then(response => response.json())
+      .then(players => { this.setState({loading: false, players: players}) })
```
Comment (patrickkenzie) [inline client/src/Components/TradeSimulator/index.js]: We could sort the players here to maintain the same ordering as the old setup.

File: client/src/Components/TradeSimulator/Chart.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import 'chartjs-plugin-annotation'
+
+export default class Chart extends Component {
+  render () {
+    const { players, teamNames, salaryCap } = this.props
+
+    const data = {
+      labels: teamNames,
+      datasets: _.flatten(teamNames.map(team => {
+        const teamPlayers = _.sortBy(players.filter((p) => p.team === team), (p) => p.salary)
```
Comment (kevinhughes27) [inline client/src/Components/TradeSimulator/Chart.js]: not super obvious but the players are still sorted its just here now.

**Comment by kevinhughes27** [comment]
I updated the spreadsheet sync script to account for the new API and the fact that stats and salaries are separated now.

**Pull request merged**